specs/interop/fault-proof.md

```diff
@@ -4,24 +4,110 @@
+  disputed L2 block.
+- The L2 block number preimage oracle local key now always provides the timestamp of the proposal.
+- The L2 chain ID preimage oracle local key is no longer used and is never populated by the dispute game.
 
-Deposits are a special case, synchronous with derivation, at enforced cross-L2 delay.
-Thus deposits cannot reference each other's events intra-block.
+## Super Root State Transition
 
-No changes to the dispute game bisection are required. The only changes required are to the fault proof program itself.
-The insight is that the fault proof program can be a superset of the state transition function.
+The super fault dispute game defines a state transition from the current anchor state super root to the canonical super
+root at the game's proposal timestamp, if one can be derived from the available L1 data or the invalid state if not.
+
+As the valid state transition always extends to the game proposal's timestamp, proposing a super root with a timestamp
+less than or greater than game's proposal timestamp will be found to be invalid. Similarly any root claim that is the
+hash of a `TransitionState` will be found to be invalid.
+
+To reduce the amount of processing required in a single invocation of the FPVM, the state transition breaks the
+transition from the super root at one timestamp to the super root at the next timestamp into 128 steps. The claims for
+these intermediate steps are `keccak256` hash of a [transition state](#transition-state). These 128 steps repeat to
+transition between the super root at each timestamp from the anchor state until the proposal time is reached.
+
+### Transition State
+
+A transition state is RLP encoded data consisting of:
+
+- `SuperOutput []byte` - the encoded super output at the immediately prior timestamp. This is the super root that is
+  being transformed _from_.
+- `PendingProgress []LocalSafeBlock` - the next block derived for each chain from L1 batch data, prior to validating
+  executing messages.
+  - `LocalSafeBlock` is two `bytes32` RLP encoded, the first being the block hash and the second the output root for
+    the block.
+- `Step` a `uint64` recording the number of steps applied to this `TransitionState` since the `SuperOutput`.
+
+### Invalid State
+
+The invalid state is defined as `keccak256(utf8("invalid"))`. This state is used to indicate a state transition where
+there is no possible valid proposal. The dispute game contract MUST prevent a game from being created where the root
+claim is the invalid state.
+
+When the prestate is the invalid state, the post state is also the invalid state.
+
+### Local Safe Block Derivation
+
+The first 127 steps derive the next local safe block for one chain in the super root using
+the [derivation process](../protocol/derivation.md). Chains are processed in the order they appear in the super root
+(ascending order of chain ID). The `Step` is incremented after each step.
```

Comment:
> ascending order of chain

This is somewhat new to me. Is this basically what the "chain index" will be?
I think it'll be less ambiguous to refer to the ordering of chains in the super output instead. Though I'd imagine in practice both orders will be the same.

Reply:
Yeah the chain index is the index of the chain in the super output (and dependency set - the ordering rules are the same). So the chain with the lowest chain ID is chain index 0.